packages/aws-appsync/src/link/offline-link.js

```diff
@@ -46,17 +46,23 @@ export class OfflineLink extends ApolloLink {
             }
 
             if (isMutation) {
-                const data = processMutation(operation, this.store);
+                const { cache, optimisticResponse, AASContext: { doIt = false } = {} } = operation.getContext();
 
-                // If we got data, it is the optimisticResponse, we send it to the observer
-                // Otherwise, we allow the mutation to continue in the link chain
-                if (data) {
-                    observer.next({ data });
-                    observer.complete();
+                if (!doIt) {
+                    if (!optimisticResponse) {
+                        console.warn('An optimisticResponse was not provided, it is required when using offline capabilities.');
 
-                    return () => null;
-                } else {
-                    // console.log('Processing mutation');
+                        if (!online) {
+                            throw new Error('Missing optimisticReponse while offline.');
```

Comment:
typo optimisticResponse

Maybe I would add a code comment something like 
// offline cache without optimistic response is processed immediately